File: Front-End/src/app/core/models/view-models/usernotification.ts
export interface UserNotification {
  isRead: boolean;
  notificationId: number;
  userId: number;
  notification: Notification;
  notificationText: string;
  notificationDate: string;
  movieID: number;
}
export interface Notification {
  notificationText: string;
  dateTime: string;
  creatorId: number;
}

File: MovieShop/Database/UserNotification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MovieShop.Database
{
    public class UserNotification
    {
        [Key]
        public int Id { get; set; }
        public bool IsRead { get; set; }

        [ForeignKey("Notification")]
        public int NotificationId { get; set; }
        public Notification Notification { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }
        public User User { get; set; }
    }
}
